src/uct/ib/rc/base/rc_ep.c

```diff
@@ -222,11 +222,16 @@ void uct_rc_ep_am_packet_dump(uct_base_iface_t *iface, uct_am_trace_type_t type,
                               char *buffer, size_t max)
 {
     uct_rc_hdr_t *rch = data;
-    uint8_t am_wo_fc = rch->am_id & ~UCT_RC_EP_FC_MASK; /* mask out FC bits*/
-
-    snprintf(buffer, max, " am %d ", am_wo_fc);
-    uct_iface_dump_am(iface, type, am_wo_fc, rch + 1, length - sizeof(*rch),
-                      buffer + strlen(buffer), max - strlen(buffer));
+    uint8_t fc_hdr    = uct_rc_fc_get_fc_hdr(rch->am_id);
+    uint8_t am_wo_fc;
+
+    /* Do not invoke AM tracer for auxiliary pure FC_GRANT message */
+    if (fc_hdr != UCT_RC_EP_FC_PURE_GRANT) {
```

Comment:
how about the following format:
`UCX  DATA  SEND qp 0x1596e --- [inl len 9] [inl len 0] am 0 (FC bits: s-)` // AM with soft credits request
`UCX  DATA  SEND qp 0x1596e --- [inl len 9] [inl len 0] am 0 (FC bits: h-)` // AM with hard credits request
`UCX  DATA  SEND qp 0x1596e --- [inl len 9] [inl len 0] am 0 (FC bits: -g)` // AM with credits grant
`UCX  DATA  SEND qp 0x1596e --- [inl len 9] [inl len 0] am 0 (FC bits: sg)` // AM with soft credits request and grant bundled together
`UCX  DATA  RECV qp 0x16f0e [va 0x7ffff460a680 len 1] FC pure grant am` // Pure FC grant